### PR TITLE
thread.h: extend doc for msg_waiters and msg_array

### PR DESCRIPTION
**Pull request by Lotterleben**
I've found the documentation for _thread.msg_waiters and _thread.msg_array to be a bit ambiguous. This PR attempts to explain what they do more clearly.

### CODE REVIEW

File: core/include/thread.h [hunk 1]
```diff
@@ -184,9 +184,12 @@ struct _thread {
...
 #endif
 #if defined(MODULE_CORE_MSG)
-    list_node_t msg_waiters;        /**< threads waiting on message     */
+    list_node_t msg_waiters;        /**< threads waiting for their message
+                                         to be delivered to this thread
+                                         (i.e. all blocking sends)      */
     cib_t msg_queue;                /**< message queue                  */
-    msg_t *msg_array;               /**< memory holding messages        */
+    msg_t *msg_array;               /**< memory holding messages sent
+                                         to this thread                 */
```
Comment (kaspar030) [inline core/include/thread.h]: Could you rephrase this as "... to this thread's message queue"?

File: core/include/thread.h [hunk 2]
```diff
@@ -184,9 +184,12 @@ struct _thread {
                                          flags                          */
 #endif
 #if defined(MODULE_CORE_MSG)
-    list_node_t msg_waiters;        /**< threads waiting on message     */
+    list_node_t msg_waiters;        /**< threads waiting for their message
+                                         to be delivered to this thread
+                                         (i.e. all blocking sends)      */
     cib_t msg_queue;                /**< message queue                  */
```
Comment (kaspar030) [inline core/include/thread.h]: Here, "index of this thread's message queue (if any)" would be to the point

File: core/include/thread.h (hunk 1 above)
Comment (Lotterleben) [inline core/include/thread.h]: Will do.

File: core/include/thread.h (hunk 2 above)
Comment (Lotterleben) [inline core/include/thread.h]: Maybe I gravely misunderstood what msg_waiters does, but I don't quite get how this describes its purpose...

Comment (kaspar030) [inline core/include/thread.h]: I meant changing the doc of msg_queue?

Comment (Lotterleben) [inline core/include/thread.h]: oooooh m) Good point, will do!

File: core/include/thread.h [hunk 3]
```diff
@@ -184,9 +184,13 @@ struct _thread {
                                          flags                          */
 #endif
 #if defined(MODULE_CORE_MSG)
-    list_node_t msg_waiters;        /**< threads waiting on message     */
-    cib_t msg_queue;                /**< message queue                  */
-    msg_t *msg_array;               /**< memory holding messages        */
+    list_node_t msg_waiters;        /**< threads waiting for their message
+                                         to be delivered to this thread
+                                         (i.e. all blocking sends)      */
```
Comment (OlegHahm) [inline core/include/thread.h]: I think "all blocked sends" would be more exact, because a blocking `msg_send()` call may deliver its message to a potential queue directly without ending up in this list.

Comment (Lotterleben) [inline core/include/thread.h]: Good point, done :)

**Comment by Lotterleben** [comment]
@kaspar030 , happy as well? :)

**Comment by Lotterleben** [comment]
@kaspar030 ping? :>

File: core/include/thread.h [hunk 4]
```diff
@@ -184,9 +184,13 @@ struct _thread {
...
 #endif
 #if defined(MODULE_CORE_MSG)
-    list_node_t msg_waiters;        /**< threads waiting on message     */
-    cib_t msg_queue;                /**< message queue                  */
-    msg_t *msg_array;               /**< memory holding messages        */
+    list_node_t msg_waiters;        /**< threads waiting for their message
+                                         to be delivered to this thread
+                                         (i.e. all blocked sends)       */
+    cib_t msg_queue;                /**< index of this thread's
+                                         message queue (if any)         */
```
Comment (miri64) [inline core/include/thread.h]: maybe add a hyperlink: `[thread's message queue](thread_t::msg_array)`.

**Comment by miri64** [comment]
Please squash

**Comment by Lotterleben** [comment]
Squashed
